icu4c/source/common/bytesinkutil.h

```diff
@@ -83,6 +86,42 @@ class U_COMMON_API CharStringByteSink : public ByteSink {
     CharString& dest_;
 };
 
+/**
+ * Calls a lambda that writes to a ByteSink with a CheckedArrayByteSink and
+ * then returns through u_terminateChars(), in order to implement the classic
+ * ICU4C C API writing to a fix sized buffer on top of a contemporary C++ API.
+ *
+ * @param buffer receiving buffer
+ * @param capacity capacity of receiving buffer
+ * @param status set to U_BUFFER_OVERFLOW_ERROR on overflow
+ * @param lambda that gets called with the sink as an argument
+ * @return number of bytes written, or needed (in case of overflow)
+ * @internal
+ */
+template <
+    typename F,
+    typename = std::enable_if_t<std::is_invocable_r_v<void, F, ByteSink&>>>
+int32_t
+u_withCheckedArrayByteSinkAndTerminateChars(
+        char* buffer,
+        int32_t capacity,
+        UErrorCode& status,
```

Comment:
It's also unusual to have the UErrorCode not as the last argument. Is this required for the template to work?

If possible, it's nicest to have
- config
- input
- output
- UErrorCode

Reply:
Done.